### DISCUSSION TITLE
Podcasts: how to use with existing tagged audio files?

### DISCUSSION THREAD

**Discussion by JakobTischler** [advplyr/audiobookshelf#1038] [Ideas]
Hey all,

so for the past few years I've been writing my own tools and scripts to download and auto-tag podcasts. Thus I have a rather large collection of podcast files that I use with Plex (not ideal, obviously), and that I'd rather like to use with ABS.

From what I've seen and tested, the podcast integration is set up for entering a feed and having ABS download the files, but no real support for existing local files. Yes, I can add the folders with its files, but the mp3 tags aren't used at all from what I can gather, and so there's no real way to create a usable library. Is that correct? If so, are there any plans to implement that?

Thanks in advance for any information.

**Comment by advplyr** [comment]
You are correct, podcast episodes scanned in are not using the meta tags in the audio files. Can you share the tags you are using for episodes?

**Comment by JakobTischler** [comment]
Sure. There's no tagging standard, really, but I'll just share what I came up with. I'm using MP3Tag, so most of these are the unified tags. There might be some that don't exist in ID3v2.3, but do in ID3v2.4 or MP4 - or vice versa.

![image](https://user-images.githubusercontent.com/748857/193668784-182ad19a-8de6-4b28-b11c-b38409230e78.png)

* `album`, `albumsort`: Podcast title
* `artist`: Hosts plus guests. I did this specifically for Plex so I could see the guests in the track view. If this should be implemented in ABS, `artist` should only be the host, with maybe `composer` as guests (similar to `composer` being used as the narrator tag for audiobooks).
* `artistsort`, `albumartist`, `albumartistsort`: Hosts
* `comment`, `unsyncedlyrics`: Episode description
* `composer`, `guests`: Guests, if applicable. `guests` is not a real tag (and should be ignored), I created and misused it for automatic file naming options. I think `performer` could be used as well.
* `contentgroup`: Kind of like a subtitle. It's a residue from an audiobook tagging standard that I continued with this. Not really necessary, really.
* `discnumber`: Season, if applicable
* `genre`: Well, the genres
* `language`: Language
* `movementname`, `series`: Podcast title
* `movement`, `series-part`, `track`: episode number
* `origyear`: publication year
* `podcast`: I think this is an iTunes specific tag that's always set to 1
* `publisher`: podcast studio
* `showmovement`: this is part of the `movement` tag, not sure which software uses that.
* `title`: episode title
* `year`: publication date. I use yyyy-mm-dd specifically for Plex, in case it ever supports that on a track level. The whole publication date thing can be juggled around as there are a handful of tags that might be valid: `year`, `date`, `releasedate`, ...

Season and episode number together with `series-part` can be tricky. But since `series-part` can be a string, and not only a number, one might also do a "season.episode" series-part, IMO (like 6.3 for season 6 episode 3=.

**Comment by Dob4115** [comment]
Was searching for exactly this discussion! I was wondering if mp3 tags were read, and now I know. I would like to second this request... without it I dont see AudioBookShelf being very helpful for my podcasts. I guess not everyone "hoards," but for those who do and have an existing collection of podcasts, this is critical. Sure, I could populate most of the data from AudioBookShelf, but then it's not transferable. I'd rather put my effort into the file metadata, suck it into AudioBookShelf, and then add the couple things that AudioBookShelf supports. Maybe, for new podcast episodes I'll have them downloaded via AudioBookShelf, but I still want the information in the file.

I also want to thank @JakobTischler for sharing his tagging schema. While more detailed than I need, I like the basics and plan on implementing this for my own podcast files.

**Comment by tfranken90** [comment]
This would be an extremely useful feature. I have a large collection of radio series which are tagged following a scheme similar to @JakobTischler Track number for episode, disc number for season, date for air date (formatted as YYYY-MM-DD), and  album artist as author or publisher where applicable (for example, Douglas Adams is the author on the Hitchhikers Guide to the Galaxy or BBC might be album artist on a more generic release). 

Rather than scanning tags, it would be helpful if ABS could just pickup some basic information from folder structure and file names. For example, my files are named with a pretty conventional SS-EE scheme. 

It's also worth noting that such radio series don't quite fit into either category. Unlike audiobooks, they have discrete episodes with discrete release dates but unlike podcasts their is no need for updating through rss feeds. They are much more like a television series than anything else.

**Comment by advplyr** [comment]
I couldn't find an open feature request for this so I created one off of @JakobTischler's post. https://github.com/advplyr/audiobookshelf/issues/1488

Please add comments/suggestions there with how this should be handled.
